src/nvim/terminal.c

```diff
@@ -1070,6 +1071,25 @@ static int term_sb_pop(int cols, VTermScreenCell *cells, void *data)
   return 1;
 }
 
+static int term_sb_clear(void *data)
+{
+  Terminal *term = data;
+
+  if (!term->sb_size || !term->sb_current) {
+    return 1;
+  }
+
+  for (size_t i = 0; i < term->sb_current; i++) {
+    xfree(term->sb_buffer[i]);
+  }
+
+  term->sb_current = 0;
+  term->sb_pending = 0;
+  refresh_terminal(term);
```

Comment:
Using `pmap_put(ptr_t)(&invalidated_terminals, term, NULL);` causes the test to fail: the first line becomes `line17` instead of `line26`. Not sure why.